src/charm.py

```diff
@@ -266,8 +271,31 @@ def _generate_du_config(self) -> str:
             tac=remote_network_config.tac,
             plmns=remote_network_config.plmns,
             simulation_mode=self._charm_config.simulation_mode,
+            frequency_band=self._charm_config.frequency_band,
+            sub_carrier_spacing=_get_numerology(self._charm_config.sub_carrier_spacing),
+            absolute_frequency_ssb=get_absolute_frequency_ssb(self._charm_config.center_frequency),
+            dl_absolute_frequency_point_a=get_dl_absolute_frequency_point_a(
+                self._charm_config.center_frequency,
+                self._charm_config.bandwidth,
+                self._charm_config.sub_carrier_spacing,
+            ),
+            dl_carrier_bandwidth=self._get_carrier_bandwidth(),
+            ul_carrier_bandwidth=self._get_carrier_bandwidth(),
+            initial_dl_bwp_location_and_bandwidth=get_initial_bwp(
+                self._get_carrier_bandwidth(),
+            ),
+            initial_ul_bwp_location_and_bandwidth=get_initial_bwp(
+                self._get_carrier_bandwidth(),
+            ),
         ).rstrip()
 
+    def _get_carrier_bandwidth(self) -> int:
+        """Return the carrier bandwidth."""
+        return get_carrier_bandwidth(
+            self._charm_config.bandwidth,
+            self._charm_config.sub_carrier_spacing,
+        )
+
```

Comment:
```suggestion
    @lru_cache
    def _get_carrier_bandwidth(self) -> int:
        """Return the carrier bandwidth."""
        return get_carrier_bandwidth(
            self._charm_config.bandwidth,
            self._charm_config.sub_carrier_spacing,
        )

```

Seeing that this gets called 4 times in a row with always the same parameter, this would be a good use of `lru_cache`, saving redundant compute. This would also need the following import:

```
from functools import lru_cache
```

Reply:
Done